builder/softlayer/step_capture_image.go

```diff
@@ -4,35 +4,81 @@ import (
 	"fmt"
 	"github.com/mitchellh/multistep"
 	"github.com/mitchellh/packer/packer"
+	"strings"
 )
 
 type stepCaptureImage struct{}
 
+func (self *stepCaptureImage) getBlockDeviceIds(blockDevices []interface{}) ([]int64) {
+	blockDeviceIds := make([]int64, len(blockDevices))
+	deviceCount := 0
+
+	for _, val := range blockDevices {
+		blockDevice := val.(map[string]interface{})
+		diskImage := blockDevice["diskImage"].(map[string]interface{})
+		name := diskImage["name"].(string)
+		id := int64(blockDevice["id"].(float64))
+
+		if !strings.Contains(name, "SWAP") {
+			blockDeviceIds[deviceCount] = id
+			deviceCount++
+		}
+	}
+
+	return blockDeviceIds[:deviceCount]
+}
+
```

Comment:
Sure. I put it in the step because it is somewhat special purpose: it doesn't communicate with the SoftLayer API like the majority of other functions in client.go, and it filters the block devices to only those that are needed by the step.

To put it in client.go, I think it will need a different name, how's "findNonSwapBlockDeviceIds"?